shared/src/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -557,6 +565,22 @@ private void _start() {
         boolean pull = isPull(config.getReplicatorType());
         boolean continuous = config.isContinuous();
 
+        C4ReplicationFilter filter = new C4ReplicationFilter() {
+            @Override
+            public boolean validationFunction(final String docID, final int flags, final long dict, final boolean isPush, final Object context)
+                throws  Exception {
+                final AbstractReplicator replicator = (AbstractReplicator) context;
+                return handler.submit(new Callable<Boolean>() {
```

Comment:
I don't think we need to submit the validationFunction call to the looper. We can just call `replicator.validationFunction(docID, flags, dict, isPush)`?